GVRf/Framework/framework/src/main/java/org/gearvrf/GVRAssetLoader.java

```diff
@@ -45,23 +43,137 @@
 import org.gearvrf.scene_objects.GVRModelSceneObject;
 import org.gearvrf.utility.FileNameUtils;
 import org.gearvrf.utility.Log;
-
 import android.content.Context;
 import android.content.res.AssetManager;
 import android.os.Environment;
 import org.gearvrf.jassimp.AiColor;
+
 /**
- * {@link GVRImporter} provides methods for importing 3D models and making them
+ * {@link GVRAssetLoader} provides methods for importing 3D models and making them
  * available through instances of {@link GVRAssimpImporter}.
  * <p>
  * Supports importing models from an application's resources (both
  * {@code assets} and {@code res/raw}) and from directories on the device's SD
  * card that the application has permission to read.
  */
-final class GVRImporter {
-    private GVRImporter() {
-    }
+public final class GVRAssetLoader implements IEventReceiver {
+    /**
+     * Listens for texture load events and raises the "onAssetLoaded"
+     * event after all textures have been loaded.
+     */
+    static public class AssetEventListener implements IAssetEvents
+    {
+        protected GVRAssetLoader       mImporter = null;
+        protected int               mNumTextures = 0;
+        protected GVRScene          mScene = null;
+        protected GVRSceneObject    mModel = null;
+        protected IAssetEvents      mUserHandler = null;
+        protected String            mFileName;
+        protected String            mErrors;
+        
+        public AssetEventListener(GVRAssetLoader importer, GVRScene scene) {
+            mScene = scene;
+            mImporter = importer;
+            mNumTextures = 0;
+        }
+        public void beginLoadTexture()
+        {
+            ++mNumTextures;
+        }
+        
+        public void beginLoadAsset(String filePath, IAssetEvents userHandler)
+        {
+            mImporter.getEventReceiver().addListener(userHandler);
+            mFileName = filePath;
+            mNumTextures = 0;
```

Comment:
Looks like this will happen in the future. It won't make a difference in the existing code.